spacetemplate/repository_blackbox_test.go

```diff
@@ -0,0 +1,230 @@
+	s.DBTestSuite.SetupSuite()
+	//s.clean = cleaner.DeleteCreatedEntities(s.DB)
+	s.spaceTemplateRepo = spacetemplate.NewRepository(s.DB)
+	require.NotNil(s.T(), s.spaceTemplateRepo)
+	s.ctx = migration.NewMigrationContext(context.Background())
+	s.DBTestSuite.PopulateDBTestSuite(s.ctx)
+}
+
+func (s *TestRepositorySuite) TearDownSuite() {
+	//s.clean()
+	s.DBTestSuite.TearDownSuite()
+}
+
+func (s *TestRepositorySuite) TestRepository_Create() {
+	//s.T().Parallel()
+	resource.Require(s.T(), resource.Database)
+
+	// This is a valid space template for tests to use.
+	description := "foobar"
+	validSt := spacetemplate.SpaceTemplate{
+		Name:        "Issue Tracking",
+		Description: &description,
+		Template:    getValidTestTemplate(),
+	}
+
+	s.T().Run("valid template", func(t *testing.T) {
+		t.Parallel()
+		expected := validSt
+		expected.Name = uuid.NewV4().String() // generate a random string
+		actual := expected
+		err := s.spaceTemplateRepo.Create(s.ctx, &actual)
+		require.Nil(t, err)
+		validateAgainstExpectedSpace(t, expected, actual)
+	})
+
+	s.T().Run("valid template with given id", func(t *testing.T) {
+		t.Parallel()
+		expected := validSt
+		expected.Name = uuid.NewV4().String() // generate a random string
+		expected.ID = uuid.NewV4()
+		actual := expected
+		err := s.spaceTemplateRepo.Create(s.ctx, &actual)
+		require.Nil(t, err)
+		validateAgainstExpectedSpace(t, expected, actual)
+	})
+
+	s.T().Run("valid template without given id", func(t *testing.T) {
+		t.Parallel()
+		expected := validSt
+		expected.Name = uuid.NewV4().String() // generate a random string
+		actual := expected
+		actual.ID = uuid.Nil
+		err := s.spaceTemplateRepo.Create(s.ctx, &actual)
+		require.Nil(t, err)
+		validateAgainstExpectedSpace(t, expected, actual)
+		assert.NotEqual(t, uuid.Nil, actual.ID)
+	})
+
+	s.T().Run("invalid template (unique name violation)", func(t *testing.T) {
+		t.Parallel()
+		uniqueName := uuid.NewV4().String() // generate a random string
+		first := validSt
+		first.Name = uniqueName
+		err := s.spaceTemplateRepo.Create(s.ctx, &first)
+		require.Nil(t, err)
+		second := validSt
+		second.Name = uniqueName
+		err = s.spaceTemplateRepo.Create(s.ctx, &second)
+		require.NotNil(t, err)
+	})
+
+	s.T().Run("invalid template (empty name)", func(t *testing.T) {
+		t.Parallel()
+		actual := validSt
+		actual.Name = ""
+		err := s.spaceTemplateRepo.Create(s.ctx, &actual)
+		require.NotNil(t, err)
+	})
+
+	s.T().Run("invalid template (empty template)", func(t *testing.T) {
+		t.Parallel()
+		actual := validSt
+		actual.Name = uuid.NewV4().String() // generate a random string
+		actual.Template = ""
+		err := s.spaceTemplateRepo.Create(s.ctx, &actual)
+		require.NotNil(t, err)
+	})
+}
+
+func (s *TestRepositorySuite) TestRepository_Load() {
+	//s.T().Parallel()
+	resource.Require(s.T(), resource.Database)
+
+	// This is a valid space template for tests to use.
+	description := "foobar"
+	validSt := spacetemplate.SpaceTemplate{
```

Comment:
I did run `make analyze-go-code` and it didn't report something. But I fixed it here: f2c43619c05832ce80daf4ce596131e3a8c2769a